test/test/compare-responses.js

```diff
@@ -0,0 +1,98 @@
+const chai = require("chai");
+const expect = chai.expect;
+const fs = require('fs')
+const path = require('path')
+const sanitizeFileName = require("sanitize-filename");
+
+const constants = require('./constants');
+const SAVED_RESPONSES_PATH =  constants.SAVED_RESPONSES_PATH;
+
+/**
+ * Remove variable values such as build time, version, etc. from response.
+ *
+ * This enables us to compare responses from tests.
+ *
+ * @param response response to sanitize
+ * @return response with variable values replaces with fixed ones
+ */
+function sanitizeResponse(response) {
+    return makeResponseValuesFixed(response, {
+        // Server information page
+        blackLabBuildTime: true,
+        blacklabVersion: true,
+
+        // Corpus information page
+        versionInfo: {
+            blackLabBuildTime: true,
+            blacklabVersion: true,
+            timeCreated: true,
+            timeModified: true
+        },
+
+        // Hits/docs response
+        summary: {
+            searchTime: true,
+            countTime: true,
+        }
+    });
+}
+
+/**
+ * Replace values in  JSON object structure with fixed ones.
+ *
+ * This enables us to compare responses from tests.
+ *
+ * @param response response to sanitize
+ * @param valuesToFix (potentially nested) object structure of values that, when found, should be set to this fixed value.
+ * @return response with variable values replaces with fixed ones
+ */
+function makeResponseValuesFixed(response, valuesToFix) {
+    const cleanedData = {};
+    for (let key in response) {
+        if (!(key in valuesToFix)) {
+            // Nothing to fix. Just copy this part.
+            cleanedData[key] = response[key];
+        } else {
+            if (typeof response[key] === 'object') {
+                // Recursively fix this part of the response
+                cleanedData[key] = makeResponseValuesFixed(response[key], valuesToFix[key]);
+            } else {
+                // Make this response value fixed
+                cleanedData[key] = "VALUE_REMOVED";
+            }
+        }
+    }
+    return cleanedData;
+}
+
+/**
+ * Either save this response if it's the first time we
+ * run this test, or compare it to the previously saved version.
+ *
+ * @param testName name of this test
+ * @param actualResponse webservice response we got (parsed JSON)
+ */
+function expectUnchanged(category, testName, actualResponse) {
+    // Remove anything that's variable (e.g. search time) from the response.
+    const sanitized = sanitizeResponse(actualResponse);
+
+    // Did we have a previous response?
+    const categoryDir = path.resolve(SAVED_RESPONSES_PATH, sanitizeFileName(category));
+    if (!fs.existsSync(categoryDir))
+        fs.mkdirSync(categoryDir);
```

Comment:
An optional change: I think you can simplify the logic here by creating the file if the `category` directory is not found and return

Reply:
I'm not sure I understand. Wouldn't that mean we'd have the file creation in two places, e.g:

```js
if (dir doesn't exist) {
  fs.mkdirSync(...)
  fs.writeFileSync(...)
} else if (file-in-dir doesn't exist) {
  fs.writeFileSync(...)
}
```

Of course you can move fs.writeFileSync(...) to a separate function, but I'm not sure this change improves readability.

Or did you mean something like this:

```js
if (dir doesn't exist or file-in-dir doesn't exist) {
  fs.mkdirSync(...)
  fs.writeFileSync(...)
}
```

However, I'm not sure if fs.mkdirSync() succeeds if the directory already existed; if it fails in that case, this doesn't work.

Reply:
yes, I see we'll have to repeat the `writeFile`, no worries then.